### DISCUSSION TITLE
How to tell if the response has already been sent in an exception handler?

### DISCUSSION THREAD

**Discussion by ojii** [encode/starlette#1537] [Q&A] [answered]
If background tasks raise exceptions, these get caught by exception handlers. Normally, we can return a response in exception handlers, but if they're triggered from background tasks, the response is already sent so we get a `RuntimeError: Caught handled exception, but response already started.`. How can I determine whether the response has already been sent in the exception handler?

**Comment by Kludex** [comment, marked as answer]
A fix for this was already merged on master. See https://github.com/encode/starlette/pull/761

**Reply by ojii** [reply]
from what I understand, that simply throws away the response, but I still can't detect if that's gonna happen or not.
